src/Magento/FunctionalTestingFramework/Test/Util/ActionMergeUtil.php

```diff
@@ -106,22 +106,28 @@ private function resolveSecretFieldAccess($resolvedActions)
         $actions = [];
         foreach ($resolvedActions as $resolvedAction) {
             $action = $resolvedAction;
-            $hasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
-
-            if ($resolvedAction->getType() !== 'fillField' && $hasSecretRef) {
-                throw new TestReferenceException("You cannot reference secret data outside of fill field actions");
+            $actionHasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
+            $approvedActions = ['fillField', 'magentoCLI', 'field'];
+            $secretMapping = ['fillField' => 'fillSecretField'];
```

Comment:
Move these two to `const` at the class level

Reply:
Done.